bpf/lib/nodeport.h

```diff
@@ -1507,6 +1543,28 @@ static __always_inline int dsr_reply_icmp4(struct __ctx_buff *ctx,
 		.tot_len	= bpf_htons((__u16)tot_len),
 	};
 
+	struct iphdr inner_ip_hdr __align_stack_8 = *ip4;
+	__be16 dport;
+	__s32 l4_dport_offset;
+
+	/*
+	 * DSR changes the destination address from service ip to pod ip and
+	 * destination port from service port to pod port. While resppnding
+	 * back with ICMP error, it is necessary to set it to original ip and
+	 * port.
+	 */
+	inner_ip_hdr.daddr = ctx_load_meta(ctx, CB_ADDR_V4);
+	inner_ip_hdr.check = csum_fold(
```

Comment:
Is that necessary to recompute du whole csum for ip hdr, can't you just make the diff with service and pod?

Reply:
Existing checksum is already fold'ed using csum_fold. So, I could not find a way to do a diff. Let me know if you have a way.

Reply:
I see, in such case you will have to unfold and add the diff. I'm just not sure what is the preference in such case.

Could you add a comment saying something like:

```
/* We do recompute the whole checksum here.
 * Another way would be to unfold checksum and
 * then do the math adding the diff. */
```

Reply:
Done.